.github/actions/build-sample-app/action.yml

```diff
@@ -62,10 +71,23 @@ runs:
     
     - name: Setup build environment to prepare for building
       shell: bash
+      env:
+        BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+        COMMIT_HASH: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.sha }}
       run: |
         make setup_sample_app app=${{ inputs.sample-app }}
-        sd CUSTOMERIO_WORKSPACE_SITE_ID ${{ inputs.customerio-workspace-siteid }} "Apps/${{ inputs.sample-app }}/BuildEnvironment.swift"
-        sd CUSTOMERIO_WORKSPACE_CDP_API_KEY ${{ inputs.customerio-workspace-cdp-api-key }} "Apps/${{ inputs.sample-app }}/BuildEnvironment.swift"
+        sd 'buildTimestamp: TimeInterval = .*' "buildTimestamp: TimeInterval = $(date +%s)" "Apps/${{ inputs.sample-app }}/BuildEnvironment.swift"
+
+        sd 'cdpApiKey: String = \".*\"' "cdpApiKey: String = \"${{ inputs.customerio-workspace-cdp-api-key }}\"" "Apps/${{ inputs.sample-app }}/BuildEnvironment.swift"
+        sd 'siteId: String = \".*\"' "siteId: String = \"${{ inputs.customerio-workspace-siteid }}\"" "Apps/${{ inputs.sample-app }}/BuildEnvironment.swift"
+        sd 'workspaceName: String = \".*\"' "workspaceName: String = \"${{ inputs.customerio-workspace-name }}\"" "Apps/${{ inputs.sample-app }}/BuildEnvironment.swift"
+        if [[ -n "${{ inputs.customerio-public-sdk-version }}" ]]; then
```

Comment:
`customerio-public-sdk-version` needs to be passed in public builds so it gets reflected in app build info